### ISSUE TITLE
Reduce used crates for public libraries

### ISSUE THREAD

**Issue by akorchyn** [near/nearcore#11652]
Currently, we have pretty huge primitive and core libraries that pull too many crates for their need.
This issue is created as part of the code review process for #11639 to track improvements.

Recently, I have been working to improve that situation in the following PRs: #11462, #11597, #11578, #11639
That said, this process is far from being completed. As part of the work I have found potential improvements that could be made:
* Extract business logic from near-primitives
* near-async uses near-o11y for logging only. This crate is potential overkill 
* near-store can feature gate metrics by that make near-o11y optional
* feature-gate serde implementation
* iterate over all libraries removing dependency for near-async where it is used only for near-time
* nightly pulls all optional dependencies. The current nightly fmt script prevents using the '?' symbol in the feature list, so it's impossible to have something like that: ![image](https://github.com/near/nearcore/assets/42449190/f083593b-aa3b-41cd-baad-46a25cc78065) The default syntax enables optional dependencies that make nightly pull extra and unused deps.

Please, feel free to extend this list, and let's track it in one place.

**Comment by Longarithm** [comment]
@akorchyn there is issue with building specific intermediate nearcore crates. When I build `near-epoch-manager`, cargo complains about missing imports, which is solved by additional features propagating (e.g. `rand`) as in #11732 (WIP). Could you look into impacted crates as well?

**Comment by Longarithm** [comment]
One more instance of that issue: #12068
near-epoch-manager is fixed now.

**Comment by VanBarbascu** [comment]
@akorchyn, is this still something that you are working on?

**Comment by akorchyn** [comment]
Nope, I'm not, but it's probably still relevant. Though, might be less relevant now as we are trying to push openapi generated types

**Comment by walnut-the-cat** [comment]
Closing it for the time being. Please re-open if it becomes relevant again
